Update commit message template to request an explanation of why a PR fixes the bug

ddkilzer opened the pull request:
#### 0e7830380b8128fecbde01e1ef76eb82148466a3
<pre>
Update commit message template to request an explanation of why a PR fixes the bug
<a href="https://bugs.webkit.org/show_bug.cgi?id=248012">https://bugs.webkit.org/show_bug.cgi?id=248012</a>
&lt;rdar://102447501&gt;

Reviewed by Ryosuke Niwa.

Add the following template text to a commit message to remind
folks to explain why the patch fixes the bug:

    Explanation of why this fixes the bug (OO-PS!).

* .github/pull_request_template.md:
- Update GitHub documentation.
* Tools/Scripts/hooks/prepare-commit-msg:
(message):
- Update the template text.

Canonical link: <a href="https://commits.webkit.org/256835@main">https://commits.webkit.org/256835@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78066a222ccef79895f78289b28f26bef4668f68

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96923 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6192 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30025 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106450 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/166737 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100900 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6411 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34920 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89313 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103148 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102595 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/4806 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83537 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31827 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86648 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/88519 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/74719 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/225 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20005 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/211 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21426 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5002 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43948 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2297 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1440 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40725 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Tools/Scripts/hooks/prepare-commit-msg:
```diff
@@ -82,6 +82,8 @@ def message(source=None, sha=None):
 
 Reviewed by NOBODY (OOPS!).
 
+Short explanation why this fixes the bug (OOPS!).
```
I support this change. Let's see what other people think.

JonWBedard replied:
This is a reasonable idea, although I think we need to get buy-in, because this is effectively a policy change more than a code change.

@ddkilzer, have we started a `webkit-dev` thread about this? I don't think we should keep this idea in limbo for long (seem like @alanbaradlay and myself, at least, will be in support), but we should give folks an opportunity to object.

ddkilzer replied:
Posted here:
<https://lists.webkit.org/pipermail/webkit-dev/2022-November/032404.html>

And here if you're on `webkit-dev` mailing list:
<[message://%3cACBF6DEA-CB23-495C-B22E-20999FFD58D6@webkit.org%3e](message://%%3cACBF6DEA-CB23-495C-B22E-20999FFD58D6@webkit.org%%3e)>

ddkilzer replied:
Feedback seems mostly positive.  No major objections.

rniwa reviewed (approved):
r=me

Commit:
Update commit message template to request an explanation of why a PR fixes the bug
https://bugs.webkit.org/show_bug.cgi?id=248012
<rdar://102447501>

Reviewed by Ryosuke Niwa.

Add the following template text to a commit message to remind
folks to explain why the patch fixes the bug:

    Explanation of why this fixes the bug (OO-PS!).

* .github/pull_request_template.md:
- Update GitHub documentation.
* Tools/Scripts/hooks/prepare-commit-msg:
(message):
- Update the template text.

Canonical link: https://commits.webkit.org/256835@main
